File: Scala-TTT/src/main/scala-2.11/com/razorthink/scala/player/Comp.scala
package com.razorthink.scala.player

import com.razorthink.scala.resources.Print
import scala.util.Random
import com.razorthink.scala.ttt.TTT
import com.razorthink.scala.resources.Resources



class Comp (compName : String) extends Player  {

	var main = TTT;
	var board =  main.board;
	var rand = new Random();
	val print = new Print;
	playerName = compName;
	def getPlayerName : String = return playerName;
	def error : Unit = if(main.debug)println("Computer Error");
	def getInput:Array[Int] = {
			if(main.debug){
				println(playerName + "'s chance : ")
			}
			var freeCells = board.getFreeCells;
			var r = rand.nextInt(freeCells.size()) + 0;
			var rc = freeCells.get(r);
			return print.parseInt(rc.charAt(0).toString()+ " "+ rc.charAt(1).toString());
	}
}

File: TicTacToe/src/main/java/com/example/games/ttt/players/Comp.java
package com.example.games.ttt.players;

import com.example.games.ttt.board.Board;

import java.util.Random;

/**
* Created by Prathyush SP on 1/13/16.
**/

public class Comp extends Player{
	Random rand=new Random();

	public int getInput()
	{
		System.out.print(player1 + "'s chance: ");
		int x=rand.nextInt(3)+0;
		int y=rand.nextInt(3)+0;
		System.out.println(x+" "+y);
		if (Board.mat[x][y] == Board.defchar)
			Board.mat[x][y] = 'X';
		else {
			error();
			return -1;
		}	
		return 0;		
	}
}